src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        // generate a specific figure based on the `figureNumber` value
+        int number = (int) (Math.random() * FIGURE_COUNT);
+        String color = colorSupplier.getRandomColor();
+        Figure randomFigure = new Circle(10);
+        double randomNumber1 = Math.floor(Math.random() * 1000 * 100) / 100.0;
+        double randomNumber2 = Math.floor(Math.random() * 1000 * 100) / 100.0;
+        double randomNumber3 = Math.floor(Math.random() * 1000 * 100) / 100.0;
+        switch (number) {
+            case 0:
+                randomFigure = new Circle(randomNumber1, color);
+                break;

```

Comment:
the same for all other cases
```suggestion
                return new Circle(randomNumber1, color);
```